torch/nn/modules/upsampling.py

```diff
@@ -32,6 +32,14 @@ class Upsample(Module):
             and output tensors are aligned, and thus preserving the values at
             those pixels. This only has effect when :attr:`mode` is
             ``'linear'``, ``'bilinear'``, or ``'trilinear'``. Default: ``False``
+        recompute_scale_factor (bool, optional): recompute the scale_factor for use in the
+            interpolation calculation. If `recompute_scale_factor` is ``True``, then
+            `scale_factor` must be passed in and `scale_factor` is used to compute the
+            output `size`. The computed output `size` will be used to infer new scales for
+            the interpolation. Note that when `scale_factor` is floating-point, it may differ
+            from the recomputed `scale_factor` due to rounding and precision issues.
+            If `recomputed_scale_factor` is ``False``, then `size` or `scale_factor` will
+            be used directly for interpolation.
```

Comment:
To make things simpler, I'd recommend limiting the scope of this PR to just the changes needed for the new interpolation mode and pushing the addition of this kwarg to a separate PR.